test/unit/associations/has-many.test.js

```diff
@@ -212,6 +212,85 @@ describe(Support.getTestDialectTeaser('hasMany'), () => {
       }
     });
   });
+
+  describe('set', async () => {
+    it('should unassociate (delete) addr1 given a non-null constraint on userId', async () => {
+      const User = current.define('User', {
+        username: DataTypes.STRING,
+        UserId: {
+          type: DataTypes.INTEGER,
+          allowNull: false,
+        },
+      });
+
+      const Address = current.define('Address', {
+        street: DataTypes.STRING,
+      });
+
+      Address.belongsTo(User, {
+        foreignKey: {
+          allowNull: false,
+        },
+      });
+      User.hasMany(Address);
```

Comment:
This belongsTo association isn't necessary for this test. I think this should be sufficient:

```suggestion
      User.hasMany(Address, {
        foreignKey: {
          allowNull: false,
        },
      });
```